File: 프로그래머스/3/12927. 야근 지수/야근 지수.py
import heapq

def solution(n, works):
    answer = 0
    
    pq = []
    for w in works:
        heapq.heappush(pq, -w)
        
    while n > 0:
        if not pq:
            break
        num = heapq.heappop(pq)
        if num + 1 < 0:
            heapq.heappush(pq, num + 1)
        n -= 1
        
    while pq:
        answer += heapq.heappop(pq) ** 2

    return answer

File: 프로그래머스/3/12927. 야근 지수/야근 지수.cpp
#include <string>
#include <vector>
#include <queue>
#include <math.h>

using namespace std;

long long solution(int n, vector<int> works) {
    long long answer = 0;
    
    priority_queue<int> pq;
    for (int w: works) {
        pq.push(w);
    }
    
    while (n--) {
        int num = pq.top();
        pq.pop();
        pq.push(num - 1 > 0 ? num - 1:0);
    }
    
    while(!pq.empty()) {
        answer += pow(pq.top(), 2);
        pq.pop();
    }
    return answer;
}